build.gradle

```diff
@@ -195,7 +196,10 @@ subprojects {
       }
     }
     // As a generous general timeout, instead of the 6h of GHA.
-    timeout = Duration.ofMinutes(15)
+    // But only on CI or longer needing debug sessions get killed by the timeout.
+    if (isCiServer) {
+      timeout = Duration.ofMinutes(15)
```

Comment:
We should probably add debug detection to the timeout extension.

https://github.com/junit-team/junit5/blob/ee52c64336ad3a079d598bd1c3884bafbc285047/junit-platform-commons/src/main/java/org/junit/platform/commons/util/RuntimeUtils.java#L42

Reply:
I used debug detections in the past and they only work semi-reliably from my experience.

But I also think it is not the right way, but like I did.

My box was occupied with other stuff and at the same time doing a full `check` on Spock and the 15 minutes were busted even without debugging. I don't see any reason to have that limit locally, you can at any time abort the build manually easily if you see it hangs.